metaflow/plugins/aws/batch/batch.py

```diff
@@ -327,6 +329,41 @@ def create_job(
                 if key in attrs:
                     k, v = sanitize_batch_tag(key, attrs.get(key))
                     job.tag(k, v)
+
+            if not isinstance(BATCH_DEFAULT_TAGS, dict):
+                raise BatchException(
+                    "The BATCH_DEFAULT_TAGS config option must be a dictionary of key-value tags."
+                )
+            
+            for name, value in BATCH_DEFAULT_TAGS.items():
+                aws_tag = {'key': name, 'value': value}
+                validate_aws_tag(aws_tag)
+                job.tag(name, value)
+
+            if step_function_tags is not None:
+                aws_tags_list = []
+                for tag in step_function_tags:
+                    key_value = tag.split("=", 1)
+                    if len(key_value) == 2:
+                        aws_tags_list.append({
+                            'key': key_value[0],
+                            'value': key_value[1]
+                            })
+                for tag in aws_tags_list:
+                    validate_aws_tag(tag)
+                    job.tag(tag['key'], tag['value'])
+
+            if aws_tags is not None:
+                if not isinstance(aws_tags, dict):
+                    raise BatchException("tags must be a dictionary of key-value tags.")
+                decorator_aws_tags_list = [
+                    {'key': key,
+                        'value': val} for key, val in aws_tags.items()
+                ]
+                for tag in decorator_aws_tags_list:
+                    validate_aws_tag(tag)
+                    job.tag(tag['key'], tag['value'])
+
```

Comment:
I didn't test this part thoroughly, but there is one significant caveat with running validations in the batch setup phase, namely: when validation fails, the metadata for the run/step/task have already been registered in this case, leading to a failed task with no logs.

An alternative approach with a better UX would be to perform batch tag validation inside the `batch_decorator.py#step_init`, which will happen before job submission. This allows for erroring out fairly quickly due to malformed tags.

Is there any issue with moving the validation and the  `BATCH_DEFAULT_TAGS` usage into the decorator completely?